File: rest-common/src/main/java/org/chronopolis/rest/entities/DepositorNode.java
package org.chronopolis.rest.entities;

import javax.persistence.Entity;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import java.io.Serializable;
import java.util.Objects;

/**
 * Entity representing a join table for Depositor/Nodes
 *
 * @author shake
 */
@Entity
@Table(name = "depositor_distribution")
public class DepositorNode extends PersistableEntity implements Serializable {

    @ManyToOne
    private Depositor depositor;

    @ManyToOne
    private Node node;

    public DepositorNode() {
    }

    public DepositorNode(Depositor depositor, Node node) {
        this.depositor = depositor;
        this.node = node;
    }

    public Depositor getDepositor() {
        return depositor;
    }

    public DepositorNode setDepositor(Depositor depositor) {
        this.depositor = depositor;
        return this;
    }

    public Node getNode() {
        return node;
    }

    public DepositorNode setNode(Node node) {
        this.node = node;
        return this;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        DepositorNode that = (DepositorNode) o;
        return Objects.equals(depositor, that.depositor) &&
                Objects.equals(node, that.node);
    }

    @Override
    public int hashCode() {
        return Objects.hash(depositor, node);
    }

    @Override
    public String toString() {
        return "DepositorNode{" +
                "depositor=" + (depositor == null ? "null" : depositor.getNamespace()) +
                ", node=" + (node == null? "null" : node.username) +
                '}';
    }
}

File: rest-entities/src/main/kotlin/org/chronopolis/rest/kot/entities/depositor/DepositorNode.kt
package org.chronopolis.rest.kot.entities.depositor

import org.chronopolis.rest.kot.entities.Node
import org.chronopolis.rest.kot.entities.PersistableEntity
import java.io.Serializable
import javax.persistence.ManyToOne

/**
 * Entity representing a join table for Depositor <-> Node
 *
 * Will see if we need to impl equals/hashCode/toString
 * Also not sure if we really need Serializable
 *
 * @property depositor The [Depositor] to join on
 * @property node The [Node] to join on
 *
 * @author shake
 */
// @Entity
// @Table(name = "depositor_distribution")
class DepositorNode(
        @ManyToOne
        var depositor: Depositor? = null,

        @ManyToOne
        var node: Node? = null
) : PersistableEntity(), Serializable